arch/xtensa/Kconfig

```diff
@@ -128,9 +128,17 @@ config XTENSA_USE_SEPARATE_IMEM
 		This separate heap will be part of the internal DRAM.  It starts right after .data
 		and ends at the configurable size given by the OPTION XTENSA_IMEM_REGION_SIZE.
 
+config XTENSA_IMEM_REGION_SIZE_MAX
+	bool "Use a maximum separate heap for internal memory"
+	default n
+	depends on XTENSA_USE_SEPARATE_IMEM
```

Comment:
Another suggestion: the `XTENSA_USE_SEPARATE_IMEM` should explicitly inform that it intends to use a separate heap region (e.g.: `XTENSA_IMM_USE_SEPARATE_HEAP`)

Reply:
Good suggestion, it is easier to understand, I have modified.